lib/https.js

```diff
@@ -237,11 +235,20 @@ exports.request = function request(options, cb) {
     options = util._extend({}, options);
   }
   options._defaultAgent = globalAgent;
-  return http.request(options, cb);
-};
+  return new ClientRequest(options, cb);
```

Comment:
Could this PR break existing code that relies on monkey patching?

Reply:
Yeah, definitely a good concern. I'm not immediately aware of modules monkeypatching the client side code but it is definitely something we need to look for before landing this.

Reply:
If anyone is monkey patching `http.request`, then `http.ClientRequest` and `_http_client.ClientRequest` also need to be monkey patched since they're all exposed and equivalent. I've seen userland code using all three.

That `https.request` currently calls `http.request` is an implementation detail (i.e. semver-patch). Users should not be relying on that.

Reply:
Just found this commit after noticing our tests for https break. Turns out our instrumentation relied on this implementation detail. :)

Reply:
It seems to be breaking GCP's Stackdriver Trace as well.

Reply:
revert for v8.x in https://github.com/nodejs/node/pull/16660